state/upgrades.go

```diff
@@ -1430,3 +1430,45 @@ func CopyMongoSpaceToHASpaceConfig(st *State) error {
 	_, err = settings.Write()
 	return errors.Annotate(err, "writing controller info")
 }
+
+// CreateMissingApplicationConfig ensures that all models have an application config in the db.
+func CreateMissingApplicationConfig(st *State) error {
+	settingsColl, settingsCloser := st.db().GetRawCollection(settingsC)
+	defer settingsCloser()
+
+	var applicationConfigIDs []struct {
+		ID string `bson:"_id"`
+	}
+	settingsColl.Find(nil).All(&applicationConfigIDs)
+
+	allIDs := set.NewStrings()
+	for _, id := range applicationConfigIDs {
+		allIDs.Add(id.ID)
+	}
+
+	appsColl, appsCloser := st.db().GetRawCollection(applicationsC)
+	defer appsCloser()
+
+	var applicationIDs []struct {
```

Comment:
these aren't really *applicationIDs" these are just "applicationNames" or something to that effect.